File: 3484-lexicographically-smallest-string-after-a-swap/3484-lexicographically-smallest-string-after-a-swap.java
class Solution {
    public String getSmallestString(String sq) {
          StringBuilder s = new StringBuilder(sq);
        for (int i = 0; i < s.length() - 1; i++) {
            if ((int) s.charAt(i) % 2 == 0 && (int) s.charAt(i + 1) % 2 == 0 && (int) s.charAt(i) > (int) s.charAt(i + 1)) {
                s = swap(i, i + 1, s);
                break;
            }
            if ((int) s.charAt(i) % 2 == 1 && (int) s.charAt(i + 1) % 2 == 1 && (int) s.charAt(i) > (int) s.charAt(i + 1)) {
                s = swap(i, i + 1, s);
                break;
            }
        }
        return s.toString();
    }

    public StringBuilder swap(int se, int e, StringBuilder s) {
        char t = s.charAt(se);
        s.setCharAt(se, s.charAt(e));
        s.setCharAt(e, t);
        return s;
    }
}

File: 3484-lexicographically-smallest-string-after-a-swap/3484-lexicographically-smallest-string-after-a-swap.go
func getSmallestString(sq string) string {
    s := []rune(sq)
	for i := 0; i < len(s)-1; i++ {
		if int(s[i])%2 == 0 && int(s[i+1])%2 == 0 && int(s[i]) > int(s[i+1]) {
			s = swap(i, i+1, s)
			break
		}
		if int(s[i])%2 == 1 && int(s[i+1])%2 == 1 && int(s[i]) > int(s[i+1]) {
			s = swap(i, i+1, s)
			break
		}
	}
	return string(s)
}

func swap(se, e int, s []rune) []rune {
	t := s[se]
	s[se] = s[e]
	s[e] = t
	return s
}